src/Infocom.php

```diff
@@ -882,6 +882,41 @@ public static function getAmortTypeName($value)
         }
     }
 
+    public static function getLogServiceName(): string
+    {
+        return 'financial';
+    }
+
+    public static function displayFullPageForItem($id, ?array $menus = null, array $options = []): void
+    {
+        $ic = new self();
+
+        Html::popHeader(self::getTypeName(), $_SERVER['PHP_SELF']);
+
+        if (isset($_GET["id"])) {
+            $ic->getFromDB($_GET["id"]);
+            $_GET["itemtype"] = $ic->fields["itemtype"];
+            $_GET["items_id"] = $ic->fields["items_id"];
+        }
+        $item = false;
+        if (isset($_GET["itemtype"]) && ($item = getItemForItemtype($_GET["itemtype"]))) {
+            if (!isset($_GET["items_id"]) || !$item->getFromDB($_GET["items_id"])) {
+                $item = false;
+            }
+        }
```

Comment:
If `$item` is false, you can throw a `\Glpi\Exception\Http\NotFoundHttpException`. It should probably be done before calling the `Html::popHeader()` method.